lib/core_extensions.rb

```diff
@@ -39,8 +39,14 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
+        record.association(klass.to_sym).association_scope.each do |what|
```

Comment:
Do you mean `record.association(klass.to_sym).association_scope.unscoped`?
No, that returns everything with the type `klass` so if klass is `:hosts` then it will returned all hosts (in Rails 3).